## metagenomics ##  

*installed only on cicuta*

1. [megahit](https://github.com/voutcn/megahit)  

MEGAHIT: An ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph  

2. [metabat2](https://bitbucket.org/berkeleylab/metabat)  

version 2.12.1  

MetaBAT, an efficient tool for accurately reconstructing single genomes from complex microbial communities 

3. [prodigal](https://github.com/hyattpd/Prodigal)  

version 2.6.2  

Fast, reliable protein-coding gene prediction for prokaryotic genomes.  

4. [pplacer](https://github.com/matsen/pplacer)

Phylogenetic placement and downstream analysis.  

5. [recycler](https://github.com/Shamir-Lab/Recycler)  

Recycler is a tool designed for extracting circular sequences from de novo assembly graphs. It can be applied on isolate as well as metagenome and plasmidome data. The circular sequences it outputs may be plasmids, phages, etc.   

`/opt/Recycler/bin/recycler.py` 

6. [k-SLAM](https://github.com/aindj/k-SLAM)  

k-SLAM ultra fast alignment and taxonomic classification of metagenomic datasets  

`SLAM`

7. [phyloFlash](https://github.com/HRGV/phyloFlash)  

phyloFlash is a pipeline to rapidly reconstruct the SSU rRNAs and explore phylogenetic composition of an Illumina (meta)genomic or transcriptomic dataset. 

`perl /opt/phyloFlash-pf3.0b1/phyloFlash.pl`
