# Phylogenetics #  

1. [mafft](http://mafft.cbrc.jp/alignment/software/)

version 7.271

MAFFT is a multiple sequence alignment program for unix-like operating systems.  
`mafft in > out`

2. [raxmlHPC](http://www.exelixis-lab.org)

version 8.2.4

Standard tool for Maximum-likelihood based phylogenetic inference.

3. [iqtree](http://www.iqtree.org/doc/)

version 1.3.11.1 built Feb  1 2016 (quite old)

Efficient phylogenomic software by maximum likelihood

4. [fasttreeMP](http://www.microbesonline.org/fasttree/)

version version 2.1.8

FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences

5. [prottest](https://github.com/ddarriba/prottest3)

version 2.1

ProtTest is a bioinformatic tool for the selection of best-fit models of aminoacid replacement for the data at hand.

6. [readal](http://trimal.cgenomics.org/use_of_the_readal_v1.2)

version 1.2rev59

readAl reads protein or nucleotide alignments in several Multiple Sequence Alignment formats, including Phylip, Fasta, Clustal, NBRF/Pir, Mega and Nexus. The program detects automatically the input format and converts the alignment to other available formats.

7. [trimal](http://trimal.cgenomics.org)

version 1.2rev59

trimAl is a tool for the automated removal of spurious sequences or poorly aligned regions from a multiple sequence alignment

`trimal`

8. [PartitionFinder](http://www.robertlanfear.com/partitionfinder/tutorial/)

PartitionFinder is software to select best-fit partitioning schemes and models of molecular evolution for phylogenetic analyses.

9. [MrBayes](http://mrbayes.sourceforge.net)

version 3.2.6

MrBayes is a program for Bayesian inference and model choice across a wide range of phylogenetic and evolutionary models. MrBayes uses Markov chain Monte Carlo (MCMC) methods to estimate the posterior distribution of model parameters.

`please contact ≈Åukasz B. before you will start using MrBayes`

10. [bmge](http://gensoft.pasteur.fr/docs/BMGE/1.0/BMGE_doc.pdf)

version 1.12

Selects regions in a multiple sequence alignment that are suited for phylogenetic inference. BMGE is able to perform biologically relevant trimming on a multiple alignment of DNA, codon or amino acid sequences.

`bmge -?`  

11. [concaterpillar.py](http://rogerlab.biochemistryandmolecularbiology.dal.ca/Software/Software.htm#Concaterpillar)  


Assessment of congruence for protein alignment concatenation.
