### List of the software ###

1. [tmux](https://github.com/tmux/tmux/wiki) 

tmux is a terminal multiplexer. It lets you switch easily between several programs in one terminal, detach them (they keep running in the background) and reattach them to a different terminal. 

[tmux cheatsheet](https://gist.github.com/MohamedAlaa/2961058)

2. [raxmlHPC](http://www.exelixis-lab.org)

version 8.2.4

Standard tool for Maximum-likelihood based phylogenetic inference.

3. [iqtree](http://www.iqtree.org/doc/)

version 1.3.11.1 built Feb  1 2016 (quite old)

Efficient phylogenomic software by maximum likelihood

4. [mafft](http://mafft.cbrc.jp/alignment/software/)

version 7.271

MAFFT is a multiple sequence alignment program for unix-like operating systems.

5. [fasttreeMP](http://www.microbesonline.org/fasttree/)

version version 2.1.8

FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences

6. [bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)

version 2.2.6

Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences.

7. [tophat](https://ccb.jhu.edu/software/tophat/index.shtml)

version 2.1.0

TopHat is a fast splice junction mapper for RNA-Seq reads.

8. [sra-toolkit](https://www.ncbi.nlm.nih.gov/books/NBK158900/)

version 2.3.5-2

The SRA Toolkit will allow you to programmatically access data housed within SRA and convert it from the SRA format to the other formats.

9. [blast](https://www.ncbi.nlm.nih.gov/books/NBK279690/)

version 2.2.31+

Basic Local Alignment Search Tool (BLAST) is probably the most popular similarity search tool. Sequence similarity searching is one of the more important bioinformatics activities and often provides the first evidence for the function of a newly sequenced gene or piece of sequence. 




